### ISSUE TITLE
Smooth learning rate transition when continue training beyond total nb. of epochs

### ISSUE THREAD

**Issue by Jubeku** [ecmwf/WeatherGenerator#1040]
### Is your feature request related to a problem? Please describe.

Continuing training beyond the original set total nb of epochs normally leads to gaps and/or learning curves of different slopes, effectively never getting back onto the previous learning curve. 

<img width="896" height="456" alt="Image" src="https://github.com/user-attachments/assets/43f3a9c5-d4a5-416a-a3da-3da1c96824d5" />

### Describe the solution you'd like

Test different implementations of smooth learning rate transitions, optionally with using the last optimizer state. 
Ideas to explore: https://arxiv.org/pdf/2106.04560

### Organisation

ECMWF
